docs/reference/mapping/types.asciidoc

```diff
@@ -118,3 +120,5 @@ include::types/dense-vector.asciidoc[]
 include::types/sparse-vector.asciidoc[]
 
 include::types/search-as-you-type.asciidoc[]
+
+include::types/array.asciidoc[]
```

Comment:
I think we've been trying to keep these nav items alphabetized (although we haven't always succeeded!)

Reply:
My reasoning here was that since an array was **not** a datatype, moving it out of the standard order might indicate that.

However, I don't think it's a huge deal either way.  Moved it back to alphabetical order with e8f6d429ec2b43a4a9e0b39177a88865576f9579